File: 프로그래머스/0/181926. 수 조작하기 1/수 조작하기 1.java
class Solution {
    public int solution(int n, String control) {
        for (char ch : control.toCharArray()) {
            if (ch == 'w') {
                n++;
            } else if (ch == 's') {
                n--;
            } else if (ch == 'd') {
                n += 10;
            } else {
                n -= 10;
            }
        }
        return n;
    }
}

File: 프로그래머스/0/181926. 수 조작하기 1/수 조작하기 1.kt
class Solution {
    fun solution(n: Int, control: String): Int = n + control.sumOf {
        when (it) {
            'w' -> 1
            's' -> -1
            'd' -> 10
            'a' -> -10
            else -> 0
        }.toInt()
    }
}
